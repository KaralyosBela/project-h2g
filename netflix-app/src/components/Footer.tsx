import classes from "./Footer.module.css";

export const Footer: React.FC = () => {
  return <div className={classes.footer}>
    <h4><span>netflix</span>roulette</h4>
  </div>;
};
