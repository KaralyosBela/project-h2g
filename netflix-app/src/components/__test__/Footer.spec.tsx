import { render, screen } from "@testing-library/react";
import { Footer } from "../Footer";

describe("Footer component", () => {
  it("Rendered footer", () => {
    const {container} = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it("h4 has a text called netflixroulette", () => {
    const { getByTestId } = render(<Footer />);
    const headerEl = getByTestId("header");
    expect(headerEl.textContent).toBe("netflixroulette");
  });

  it("footer has a classname of footer", () => {
    const { getByTestId } = render(<Footer />);
    const footerEl = getByTestId("footer");
    expect(footerEl.className).toBe("footer");
  });

  it("span has a text called netflix", () => {
    const { getByTestId } = render(<Footer />);
    const spanEl = getByTestId("span");
    expect(spanEl.textContent).toBe("netflix");
  });
});
