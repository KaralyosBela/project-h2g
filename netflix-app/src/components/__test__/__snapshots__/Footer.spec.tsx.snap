// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Rendered footer 1`] = `
<div>
  <div
    class="footer"
    data-testid="footer"
  >
    <h4
      data-testid="header"
    >
      <span
        data-testid="span"
      >
        netflix
      </span>
      roulette
    </h4>
  </div>
</div>
`;
